### PR TITLE
Implement internal/errors/worker test

### PR DESCRIPTION
**Pull request by kevindiu**
### Description:

This PR implements internal/errors/worker error.
Grammar check passed

### Environment:

- Go Version: 1.15.7
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.13.1

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by hlts2 — APPROVED** [review]
LGTM

File: internal/errors/worker_test.go [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+	}
+	defaultCheckFunc := func(w want, err error) error {
+		if !Is(err, w.err) {
+			return Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "return ErrWorkerIsNotRunning error when name is index",
```
Comment (vankichi) [inline internal/errors/worker_test.go:50]:
```suggestion
			name: "return an ErrWorkerIsNotRunning error when name is index",
```

File: internal/errors/worker_test.go [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+			args: args{
+				name: "index",
+			},
+			want: want{
+				err: New("worker index is not running"),
+			},
+		},
+		{
+			name: "return ErrWorkerIsNotRunning error when name is empty",
```
Comment (vankichi) [inline internal/errors/worker_test.go:59]:
```suggestion
			name: "return an ErrWorkerIsNotRunning error when name is empty",
```

File: internal/errors/worker_test.go [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+			}
+			if test.checkFunc == nil {
+				test.checkFunc = defaultCheckFunc
+			}
+
+			err := ErrWorkerIsNotRunning(test.args.name)
+			if err := test.checkFunc(test.want, err); err != nil {
+				tt.Errorf("error = %v", err)
+			}
+
```
Comment (vankichi) [inline internal/errors/worker_test.go:88]:
```suggestion
```

File: internal/errors/worker_test.go [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+	}
+	defaultCheckFunc := func(w want, err error) error {
+		if !Is(err, w.err) {
+			return Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "return ErrWorkerIsAlreadyRunning error when name is index",
```
Comment (vankichi) [inline internal/errors/worker_test.go:117]:
```suggestion
			name: "return an ErrWorkerIsAlreadyRunning error when name is index",
```

File: internal/errors/worker_test.go [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+			args: args{
+				name: "index",
+			},
+			want: want{
+				err: New("worker index is already running"),
+			},
+		},
+		{
+			name: "return ErrWorkerIsAlreadyRunning error when name is empty",
```
Comment (vankichi) [inline internal/errors/worker_test.go:126]:
```suggestion
			name: "return an ErrWorkerIsAlreadyRunning error when name is empty",
```

File: internal/errors/worker_test.go [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+			}
+			if test.checkFunc == nil {
+				test.checkFunc = defaultCheckFunc
+			}
+
+			err := ErrWorkerIsAlreadyRunning(test.args.name)
+			if err := test.checkFunc(test.want, err); err != nil {
+				tt.Errorf("error = %v", err)
+			}
+
```
Comment (vankichi) [inline internal/errors/worker_test.go:155]:
```suggestion
```

File: internal/errors/worker_test.go [hunk 7]
```diff
@@ -0,0 +1,311 @@
...
+			}
+			if test.checkFunc == nil {
+				test.checkFunc = defaultCheckFunc
+			}
+
+			err := ErrQueueIsNotRunning()
+			if err := test.checkFunc(test.want, err); err != nil {
+				tt.Errorf("error = %v", err)
+			}
+
```
Comment (vankichi) [inline internal/errors/worker_test.go:206]:
```suggestion
```

File: internal/errors/worker_test.go [hunk 8]
```diff
@@ -0,0 +1,311 @@
...
+	}
+	defaultCheckFunc := func(w want, err error) error {
+		if !Is(err, w.err) {
+			return Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "return ErrQueueIsNotRunning error",
```
Comment (vankichi) [inline internal/errors/worker_test.go:180]:
```suggestion
			name: "return an ErrQueueIsNotRunning error",
```

File: internal/errors/worker_test.go [hunk 9]
```diff
@@ -0,0 +1,311 @@
...
+	}
+	defaultCheckFunc := func(w want, err error) error {
+		if !Is(err, w.err) {
+			return Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "return ErrQueueIsAlreadyRunning error",
```
Comment (vankichi) [inline internal/errors/worker_test.go:231]:
```suggestion
			name: "return an ErrQueueIsAlreadyRunning error",
```

File: internal/errors/worker_test.go [hunk 10]
```diff
@@ -0,0 +1,311 @@
...
+			}
+			if test.checkFunc == nil {
+				test.checkFunc = defaultCheckFunc
+			}
+
+			err := ErrQueueIsAlreadyRunning()
+			if err := test.checkFunc(test.want, err); err != nil {
+				tt.Errorf("error = %v", err)
+			}
+
```
Comment (vankichi) [inline internal/errors/worker_test.go:257]:
```suggestion
```

File: internal/errors/worker_test.go [hunk 11]
```diff
@@ -0,0 +1,311 @@
...
+			}
+			if test.checkFunc == nil {
+				test.checkFunc = defaultCheckFunc
+			}
+
+			err := ErrJobFuncIsNil()
+			if err := test.checkFunc(test.want, err); err != nil {
+				tt.Errorf("error = %v", err)
+			}
+
```
Comment (vankichi) [inline internal/errors/worker_test.go:308]:
```suggestion
```

File: internal/errors/worker_test.go [hunk 12]
```diff
@@ -0,0 +1,311 @@
...
+	}
+	defaultCheckFunc := func(w want, err error) error {
+		if !Is(err, w.err) {
+			return Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "return ErrJobFuncIsNil error",
```
Comment (vankichi) [inline internal/errors/worker_test.go:282]:
```suggestion
			name: "return an ErrJobFuncIsNil error",
```

**Review by vankichi — CHANGES_REQUESTED** [review]
please  check these suggestions

File: internal/errors/worker_test.go (hunk 1 above)
Comment (kevindiu) [reply internal/errors/worker_test.go:50]: I think the current one is correct. What do you think?

Comment (vankichi) [reply internal/errors/worker_test.go:50]: what's the current one?

Comment (kevindiu) [reply internal/errors/worker_test.go:50]: this one.
`return ErrWorkerIsNotRunning error when name is index`

Comment (vankichi) [reply internal/errors/worker_test.go:50]: no article

**Review by vankichi — APPROVED** [review]
LGTM
please get approve from @hlts2 thanks.

**Review by hlts2 — APPROVED** [review]
LGTM

**Review by vankichi — APPROVED** [review]

**Pull request merged**